code/mob/melee_attack_procs.dm

```diff
@@ -93,6 +116,29 @@
 				src.visible_message("<span class='notice'>[src] pats themselves on the back. Feel better, [src].</span>")
 
 	else
+		var/mob/living/M = target
+
+		var/obj/item/implant/projectile/dart/P = locate(/obj/item/implant/projectile/dart) in M.implant
+
+		if (P?.barbed == FALSE)
+			src.visible_message("<span class='alert'><B>[src] pulls a dart out of [target]!</B></span>")
+			P.on_remove(M)
+			M.implant.Remove(P)
+			P.set_loc(M.loc)
```

Comment:
would rather this put it in the remover's hand; helps it feel like you're actually pulling it out

Reply:
I tried using the Attackhand proc for items for putting it in hand, seems it plays the sound made when you pick up something, if that works.